clang/include/clang/AST/PreorderAST.h

```diff
@@ -25,24 +25,29 @@ namespace clang {
   using Result = Lexicographic::Result;
 
   // Each binary operator of an expression results in a new node of the
-  // PreorderAST. Each node contains 3 fields:
+  // PreorderAST. Each node contains the following fields:
+
   // Opc: The opcode of the operator.
   // Vars: A list of variables in the sub expression.
   // Const: Constants of the sub expression are folded.
+  // Others: A list of all other non-variable, non-constant expressions.
```

Comment:
I think it would be better in the long run to generalize the node type into a class hierarchy.  Something along the lines of:
```
Node
 |  \ 
 |    \
 |      \
 |        \
 |          \
BinaryNode  LeafExprNode
```
where `LeafExpr` could just wrap a clang AST expression.  A representation that separates binary expressions and regular clang expression loses the order of evaluation information, which we'll want later on.